new builtin package: py-biobb-common

d-beltran opened the pull request:
Hi everyone,

I would like to add this new package to the official repo.

Best regards

d-beltran committed:
new builtin package: biobb-common

d-beltran committed:
removed whitespace for the style test

d-beltran committed:
replaced ' by " for the style test

d-beltran committed:
changed import line for the style test

alalazo commented on var/spack/repos/builtin/packages/biobb-common/package.py:
```diff
@@ -0,0 +1,33 @@
+class BiobbCommon(PythonPackage):
+    """Biobb_common is the base package required to use the biobb packages"""
+
+    # Homepage and download url
```
```suggestion
```

alalazo commented on var/spack/repos/builtin/packages/biobb-common/package.py:
```diff
@@ -0,0 +1,33 @@
+    homepage = "https://github.com/bioexcel/biobb_common"
+    git = "https://github.com/bioexcel/biobb_common.git"
+
+    # Set the gitlab accounts of this package maintainers
+    maintainers = ["d-beltran"]
```
```suggestion
    maintainers("d-beltran")
```

d-beltran committed:
Update var/spack/repos/builtin/packages/biobb-common/package.py

Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

d-beltran committed:
Update var/spack/repos/builtin/packages/biobb-common/package.py

Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

adamjstewart commented on var/spack/repos/builtin/packages/biobb-common/package.py:
```diff
@@ -0,0 +1,31 @@
+    """Biobb_common is the base package required to use the biobb packages"""
+
+    homepage = "https://github.com/bioexcel/biobb_common"
+    git = "https://github.com/bioexcel/biobb_common.git"
```
It's best to download stable releases from PyPI: https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#downloading

d-beltran replied:
done

adamjstewart commented on var/spack/repos/builtin/packages/biobb-common/package.py:
```diff
@@ -0,0 +1,31 @@
+from spack.package import *
+
+
+class BiobbCommon(PythonPackage):
```
Is this a command-line tool or a Python API? See https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#pythonpackage-vs-packages-that-use-python

d-beltran replied:
It is both.
Every biobb (this is just the first from a long list) is prepared to run both in command line and from python.
They are actually also prepared to run in docker, singularity, cwl and galaxy.

adamjstewart replied:
We generally prefer to prefix Python libraries with `py-` unless they are predominantly used from the command line. For example, mercurial is technically a Python library, but that's not important to most users, so the package is called `mercurial`. As a counterexample, numpy as command-line tools, but most people don't know or care about that, they just want to import it in their Python code, so the package is called `py-numpy`.

d-beltran replied:
done

adamjstewart commented on var/spack/repos/builtin/packages/biobb-common/package.py:
```diff
@@ -0,0 +1,31 @@
+    version("4.0.0", branch="master")
+
+    # Dependencies
+    depends_on("py-setuptools")
```
```suggestion
    depends_on("py-setuptools", type="build")
```

adamjstewart commented on var/spack/repos/builtin/packages/biobb-common/package.py:
```diff
@@ -0,0 +1,31 @@
+    depends_on("python@3.8:", type=("build", "run"))
+    depends_on("py-pyyaml", type=("build", "run"))
+    depends_on("py-requests", type=("build", "run"))
+    depends_on("py-biopython@1.78:1.80", type=("build", "run"))
```
Where do you see these version requirements documented?

d-beltran replied:
Dependencies are stated here: https://mmb.irbbarcelona.org/biobb/documentation/source
If you click in the '+' button in the biobb-common row and then hover the '?' icon next to 'Dependencies Pypi' pr Dependencies Conda' they are shown in a tooltip.

adamjstewart replied:
I don't see any version specified for biopython...

d-beltran replied:
You are right. I inherited this restriction from an old script were it was useful to avoid some bugs but it should be no longer required.
I have removed it.

adamjstewart commented on var/spack/repos/builtin/packages/biobb-common/package.py:
```diff
@@ -0,0 +1,31 @@
+    depends_on("py-requests", type=("build", "run"))
+    depends_on("py-biopython@1.78:1.80", type=("build", "run"))
+
+    # Test
```
You can remove this, it's already automatically handled by PythonPackage

d-beltran replied:
done :)

d-beltran committed:
Update var/spack/repos/builtin/packages/biobb-common/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

d-beltran committed:
removed test

d-beltran committed:
using pypi instead of git

I hope this is declared the right way.
I tried to test it locally but I am having problems with a dependency (openblas) thus I can not even try if this works.

d-beltran committed:
upgraded version

d-beltran committed:
Added py prefix

d-beltran committed:
removed biopython version restriction

d-beltran committed:
directory with new py-prefixed name

d-beltran committed:
Delete old non-prefixed package

adamjstewart commented on var/spack/repos/builtin/packages/py-biobb-common/package.py:
```diff
@@ -0,0 +1,24 @@
+
+    # Dependencies
+    depends_on("py-setuptools", type="build")
+    depends_on("python@3.8:3.10", type=("build", "run"))
```
```suggestion
    depends_on("python@3.8:", type=("build", "run"))
```
I don't see any evidence that 3.11 doesn't work, do you?

d-beltran replied:
Acordint to [the docs](https://mmb.irbbarcelona.org/biobb/availability/python-compatibility) python 3.11 is not maintained

adamjstewart replied:
Those are just the versions that are tested. It's a pure-Python library, so it's unlikely that the minor changes between Python releases would break something. Unless you know it doesn't work with 3.11, I would suggest the above change. But you're maintaining this recipe, so it's your final call.

adamjstewart commented:
Note that #42078 also adds this package

d-beltran committed:
Update var/spack/repos/builtin/packages/py-biobb-common/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>
